docs/howto/source.rst

```diff
@@ -0,0 +1,144 @@
+To change that, we first declare for which metric we want to send values:
+
+.. code-block:: python
+
+    import metricq
+
+    class DummySource(metricq.Source):
+        ... # as above
+
+        @metricq.rpc_handler("config")
+        async def _on_config(self, **config):
+            print(f"Received new configuration: {config}")
+            self._rate = config["rate"]
+
+            metadata = {
+                "rate": rate,
+                "description": "A simple dummy metric providing random values, sent from a python DummySource",
+            }
+
+            await self.declare_metrics({"example.py.dummy": metadata})
+
+If we were in a hurry, we could omit the metadata for our metric,
+see :meth:`Source.declare_metrics` on how to do that.
+
+To finally send some values, we override :meth:`Source.task`.
+This method gets called once our Source is connected and received its initial configuration:
+
+.. code-block:: python
+
+    import metricq
+    import asyncio
+    import random
+
+    class DummySource(metricq.Source):
+        ... # as above
+
+        async def task(self):
+            while True:
+                await self.send(
+                    "example.py.dummy",
+                    time=metricq.Timestamp.now(),
+                    value=random.random(),
+                )
+                # Convert from rate (in Hz) to duration between sends (in seconds)
+                await asyncio.sleep(1 / self._rate)
+
+Improving constant-rate sources: using :class:`IntervalSource`
+--------------------------------------------------------------
+
+The above situation where we send values at a fixed rate is so common
+that we can use the convenience class :class:`IntervalSource`, which
+does all the heavy lifting for us.
+
+Note:
+    We strongly recommend implementing a :class:`IntervalSource` over a plain :class:`Source` if possible.
+    It tries to automatically compensate some timing-related issues
+    that inevitably arise in more complicated setups.
+    See the documentation for :class:`IntervalSource` for more information.
+
+To adapt the above example, we simply set :class:`IntervalSource.period` to the period of time between consecutive updates and replace :class:`Source.task` with :class:`IntervalSource.update`,
+which gets called at a constant rate:
+
+
+.. code-block:: python
+
+    import metricq
+    import asyncio
+    import random
+
+    class DummySource(metricq.Source):
+
+        @metricq.rpc_handler("config")
+        async def _on_config(self, **config):
+            rate = config["rate"]
+            # Set the update period
+            self.period = 1 / rate
+
+            await self.declare_metrics(["test.py.interval-dummy"])
+
+
+        async def update(self):
+            await self.send(
+                "example.py.interval-dummy",
+                time=metricq.Timestamp.now(),
+                value=random.random(),
+            )
```

Comment:
Please add a complete example and running a source section. The latter should include that the config comes from the manager/couchdb.